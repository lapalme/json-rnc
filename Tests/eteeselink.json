[
  {
    "books": [
      {
        "title": "Mijn leuk wereldje",
        "author": "Richard Scarry",
        "ISBN": "978-9024380329",
        "weight": 112,
        "type": "Paperback"
      },
      {
        "title": "Principia Discordia",
        "subtitle": "Or, How I Found Goddess and What I Did to Her When I Found Her: The Magnum Opiate of Malaclypse the Younger",
        "author": "Malaclypse the Younger",
        "ISBN": "978-1559500401",
        "weight": 130.4,
        "type": "Hardcover",
        "number": 48,
        "$id":  "C4567"
      }
    ],
    "owner": "George Clooney"
  },
  {
    "books": [],
    "owner": "George Bush"
  },
  {
    "name": "Amazon",
    "url": "http://www.amazon.com"
  },
  {
    "name": "Lulu",
    "url": "http://www.lulu.com"
  }
]
