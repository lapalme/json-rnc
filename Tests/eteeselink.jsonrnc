## adaptation of the "contrived example" of relax-json given at
##     https://github.com/eteeselink/relax-json

start = [BookList | Store]

BookList = { books: [ Book ], owner: string }

Book = {
  title: string, subtitle?: string, author: string,
  ISBN: string,  weight: number,    type: BookType,
  # add keys with 'special' names
  "number"?: integer, "$id"?: string 
}

Store = { name: string, url: string }

BookType = /Paperback/ | /Hardcover/
