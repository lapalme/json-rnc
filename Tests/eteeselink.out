All 1 objects are valid
